tests/rados/test_No_Cephadm_Stray_Daemon.py

```diff
@@ -0,0 +1,146 @@
+4. Ensure the OSD no longer appears in the cluster via `get_osd_list()` output.
+5. Verify no `CEPHADM_STRAY_DAEMON` warnings are raised post-removal.
+The test ensures the OSD is cleanly removed from the cluster's control plane and
+that no residual daemon artifacts or health warnings persist.
+Cluster health is verified at the end of the test to confirm it returns to `HEALTH_OK`.
+"""
+
+import json
+import random
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    Test Case: Verify that there are no CEPHADM_STRAY_DAEMON warnings while replacing the OSD
+    BZ's: https://bugzilla.redhat.com/show_bug.cgi?id=2269003, https://bugzilla.redhat.com/show_bug.cgi?id=2355037
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+
+    try:
+
+        def verify_osd_removal(osd_id: int):
+            log.info("Starting verification for removal of osd.{0}".format(osd_id))
+
+            # Step 1: Mark OSD out
+            log.info("Step 2: Marking osd out")
+            if not rados_obj.update_osd_state_on_cluster(osd_id=osd_id, state="out"):
+                err_msg = "Failed to mark osd.{0} out".format(osd_id)
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+            # Step 2: Stop OSD
+            log.info("Step 3: Stopping OSD process")
+            try:
+                rados_obj.run_ceph_command(cmd="ceph osd stop {0}".format(osd_id))
+            except Exception as e:
+                err_msg = "Failed to stop OSD.{0} process: {1}".format(osd_id, e)
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+            # Step 3: Mark OSD down
+            log.info("Step 1: Marking osd down")
+            if not rados_obj.update_osd_state_on_cluster(osd_id=osd_id, state="down"):
+                err_msg = "Failed to mark osd.{0} down".format(osd_id)
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+            # Step 4: Remove OSD via orchestrator
+            log.info("Step 4: Removing OSD with zap and replace")
+            try:
+                rados_obj.run_ceph_command(
+                    cmd="ceph orch osd rm {0} --zap --replace --force".format(osd_id)
+                )
+            except Exception as e:
+                err_msg = "Failed to remove OSD.{0} using orchestrator: {1}".format(
+                    osd_id, e
+                )
+                log.error(err_msg)
+                raise Exception(err_msg)
+
+            # Step 5: Wait and verify OSD list
+            log.info("Waiting 60 seconds before verifying updated OSD list")
+            time.sleep(60)
+            osd_ids_in_tree = rados_obj.get_osd_list()
+
+            if osd_id in osd_ids_in_tree:
```

Comment:
please update logic based on the o/p of `osd_ids_in_tree = rados_obj.get_osd_list()`